Shooter Mk 2

AustinFortunato opened the pull request:
Shooter, better, faster, stronger- brought to you by mitsubishi

AustinFortunato committed:
Cleanup for Shooter

It look real pretty now.

AustinFortunato committed:
Did more maths, and made default distance like 160 just incase

AustinFortunato committed:
CalculateRPM has a fail safe

just in case everything fails, we have a guess.

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -10,35 +10,36 @@
     public final static double THETA = Math.toRadians(37);
-    public final static double DIAMETER = 4;
-    public final static double CIRCUMFERENCE = DIAMETER * Math.PI * Math.PI;
+    public final static double BALL_SPEED_RATIO = 27.358;
```
Where'd this number come from?

AustinFortunato replied:
Basically gear ratios

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -52,24 +53,28 @@ public CommandBase spinDown() {
-        if (distanceToTarget * Math.tan(THETA) >= verticalDistance) {
-            final double gravitySide = GRAVITY * distanceToTarget * distanceToTarget;
-            final double tanSide = distanceToTarget * Math.tan(THETA) - verticalDistance;
+        if (horizontalDistance * Math.tan(THETA) >= verticalDistance) {
```
Where is `horizontalDistance` being set? It looks like it's being set once in the constructor, and not updating. Also `verticalDistance` and `distanceToTarget`, which is honestly the one that you probably need updated regularly.

AustinFortunato replied:
verticalDistance is a constant otherwise yes i will do those

AustinFortunato committed:
Updates variables from vision

     |\_/|                  
     | @ @   Woof! 
     |   <>              _  
     |  _/\------____ ((| |))
     |               `--' |   
 ____|_       ___|   |___.' 
/_/_____/____/_______|

msoucy reviewed (approved):
:ship: it
